docs/proposals/kubeedgeadm-scope.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Kubeedgeadm kubeEdge installer scope (Issue 324)
+status: Pending
+authors:
+    - "@samy2019"
+approvers:
+  - "@m1093782566"
+  - "@rohitsardesai83"
+  - "@sids-b"
+creation-date: 2019-04-11
+last-updated: 
+---
+
+## Motivation
+Many users complained that kubeEdge installation is too complicated and it may prevent people from trying kubeEdge.
+There should a simplified way to have **Getting Started with KubeEdge**, so that user can concentrate more on using it instantly, rather than
+getting entangled in the installation steps.
+
+## Proposal
+KubeEdge shall have a simple commands and steps to bring up both cloud and edge components.
+The user experience in **Getting Started with KubeEdge** will be seemless.
+Hence proposing the following commands for KubeEdge installation process.
+
+For cloud, commands shall be:
+
+- `kubeedge cloud init`
+- `kubeedge cloud reset`
+
+For edge, commands shall be:
+
+- `kubeedge node init`
+- `kubeedge node reset`
+- `kubeedge node join`
+
+**NOTE:**
+1. `node` key is used for edge component in the command, for superficial reasons. Because `kubeedge edge init` had `edge` used twice and didn't sound nice.
+2. GO will **NOT** be installed with any of the commands or processes below. User herself/himself needs to do the same in the target environment. Also it is understood and assumed that user has already build edge-controller and edge_core using the build steps mentioned in README.md
+
+## Scope of commands
+
+**Cloud commands:**
+
+`kubeedge cloud init` 
+  - What is it?
+     * This command will be responsible to bring up KubeEdge cloud components like edge-controller and K8S (using kubeadm)
+   
+  - What shall be its scope ?
+    1. Check version of OS and install subsequently the required pre-requisites using supported steps. Currently we will support **ONLY** (Ubuntu & CentOS)
+    2. Check and install all the pre-requisites before executing edge-controller, which are
+        * docker (currently 18.06.0ce3-0~ubuntu) and check if service is up
+        * kubelet, kubeadm & kubectl (latest version)
+        * openssl (latest available in OS repos)
+    3. Generate certificates using openssl and save the certs in a predefined static path.
+    It will also compress the folder and display on the terminal so that user can pick it up and transfer it to edge node (VM/host) manually.
```

Comment:
Is this compression step needed ? we can display the path of the certificates post install and give some steps for the client to follow.

Reply:
Added compression step, so that the user has to transfer only one file to target edge node. Hence reducing some effort to transfer selectively. 
Path will be displayed as mentioned here. 
Steps to transfer the file and un-compress it, i am sure the user will know how to do it.
Regarding how to use those files in edge node, the edge command will take of its usage.

Reply:
i still feel this is a burden for the installer to additionally compress , it should be ok for a user to do this. 
thoughts from others are welcome.